rust/src/common.rs

```diff
@@ -10,3 +13,14 @@ macro_rules! take_until_and_consume (
     }
   );
 );
+
+/// Free a CString allocated by Rust
+#[no_mangle]
+pub extern "C" fn rs_cstring_free(s: *mut c_char) {
```

Comment:
Should we mark the function unsafe? Clippy does complain about this.

Reply:
I think it is a good idea to follow clippy, so I'll mark the function unsafe